### PR TITLE
[receiver/webhookevent] Enable component

### PR DESCRIPTION
**Pull request by shalper2**
**Description:** Graduate webhookeventreceiver component to Alpha

**Link to tracking Issue:** [18101](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18101)

**Testing:** Ran `make otelcontribcol` and ran command with the following config:
```yaml
receivers:
  webhookevent:
    endpoint: "127.0.0.1:8088"
    read_timeout: "500ms"
    write_timeout: "500ms"
    path: "/event1"
    health_path: "/event1/healthcheck"

exporters:
  logging:

service:
  pipelines:
    logs:
      receivers: [webhookevent]
      exporters: [logging]
```
and received the following output:

```
./opentelemetry-collector-contrib/bin/otelcontribcol_darwin_amd64 --config=file:config.yaml
2023-06-08T14:13:47.036-0500	info	service/telemetry.go:104	Setting up own telemetry...
2023-06-08T14:13:47.036-0500	info	service/telemetry.go:127	Serving Prometheus metrics	{"address": ":8888", "level": "Basic"}
2023-06-08T14:13:47.036-0500	info	exporter@v0.79.0/exporter.go:275	Development component. May change in the future.	{"kind": "exporter", "data_type": "logs", "name": "logging"}
2023-06-08T14:13:47.038-0500	info	service/service.go:131	Starting otelcontribcol...	{"Version": "0.79.0-dev", "NumCPU": 12}
2023-06-08T14:13:47.038-0500	info	extensions/extensions.go:30	Starting extensions...
2023-06-08T14:13:47.038-0500	info	service/service.go:148	Everything is ready. Begin running and processing data.
```
indicating that the collector was able to successfully start with the component configured.

### CODE REVIEW

**Review by atoulme — APPROVED** [review]

File: receiver/webhookeventreceiver/metadata.yaml [hunk 1]
```diff
@@ -3,6 +3,6 @@ type: webhookevent
 status:
   class: receiver
   stability:
-    development: [logs]
-  distributions:  
+    alpha: [logs]
+  distributions: [contrib]
```
Comment (dmitryax) [inline receiver/webhookeventreceiver/metadata.yaml:7]: To add contrib here, it needs to be added in the https://github.com/open-telemetry/opentelemetry-collector-releases/blob/main/distributions/otelcol-contrib/manifest.yaml first

File: .chloggen/webhookevent-enable.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: 'bug_fix'
```
Comment (dmitryax) [inline .chloggen/webhookevent-enable.yaml:6]: I'd say it's `enhancement`. There were no bugs here

File: .chloggen/webhookevent-enable.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: webhookeventreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Wiring webhookeventreceiver by adding to component.go
```
Comment (dmitryax) [inline .chloggen/webhookevent-enable.yaml:12]:
```suggestion
note: Mark the component as Alpha and add to` component.go`
```

File: receiver/webhookeventreceiver/metadata.yaml (hunk 1 above)
Comment (shalper2) [reply receiver/webhookeventreceiver/metadata.yaml:7]: gotcha, removing contrib!

File: .chloggen/webhookevent-enable.yaml (hunk 2 above)
Comment (shalper2) [reply .chloggen/webhookevent-enable.yaml:6]: Heard! Was just going off of previous PR's for wiring, sorry!

Curious if the connecting/wiring process is documented in any official capacity or detail? I'd be happy to help with this if not

**Comment by dmitryax** [comment]
Please run `make generate` to fix the CI checks

File: .chloggen/webhookevent-enable.yaml (hunk 2 above)
Comment (dmitryax) [reply .chloggen/webhookevent-enable.yaml]: I't mentioned in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-new-components but not very detailed

**Comment by dmitryax** [comment]
The lifecycle tests are failing. Probably you need to check if `server != nil` in the Shutdown method

File: receiver/webhookeventreceiver/receiver.go [hunk 4]
```diff
@@ -133,7 +133,13 @@ func (er *eventReceiver) Start(_ context.Context, host component.Host) error {
...
 // Shutdown function manages receiver shutdown tasks. part of the receiver.Logs interface.
 func (er *eventReceiver) Shutdown(_ context.Context) error {
-	err := er.server.Close()
+    var err error
+
+    // server must exist to be closed.
+    if er.server != nil {
+	    err = er.server.Close()
+    }
+
```
Comment (codeboten) [inline receiver/webhookeventreceiver/receiver.go:142]:
```suggestion
    if er.server == nil {
        return nil
    }
    err := er.server.Close()

```

**Review by codeboten — COMMENTED** [review]
just one suggestion, there looks like there might be formatting issues in the receiver.go changes

**Pull request merged**